arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -443,7 +443,7 @@ arangodb::Result SynchronizeShard::getReadLock(
       std::unordered_map<std::string, std::string>(), timeout);
 
     auto result = putres->result;
-    if (result != nullptr && result->getHttpReturnCode() == 200) {
+    if (result != nullptr || result->getHttpReturnCode() == 200) {
```

Comment:
this is wrong now.

Reply:
Oh crap, yes, you are quite right. Bad search/replace.